### ISSUE TITLE
Stargus doesn't pick up data folder

### ISSUE THREAD

**Issue by ThibaultLemaire** [Wargus/stargus#36]
... and doesn't prompt for one.

Hi everyone!

I'm trying to make stargus run on Arch. 

So far, I've managed to compile stargus, and to extract 545.5MB of data from a Starcraft Anthology wine install (🎉).

The issue is when I start stargus, I get this :

![Stargus data was not extracted yet.][launcherror]

This doesn't match the behaviour described in the readme :

> [Upon first launch of Stargus, it will ask you for your Starcraft installation to extract the data to work with Stargus.][installinstructions]

Anyway, digging my way through the open issues, I've managed to figure out how to use `startool` and to gather some hints as to where to put the extracted data so that stargus finds it.

Following are the different locations I've tried, but none of them worked, so I'm at loss right now.

- `data/` (relative to `stargus` executable location)
- `~/.stratagus/data`
- `~/.stratagus/data.Stargus`
- `~/.stratagus/data.stargus`
- `~/.stratagus/data.sc`
- `~/.stratagus/sc`
- `~/.stratagus/Stargus`
- `~/.stratagus/stargus`

I realize most of them are pretty lame attempts, as peeking into the code of [`stargus.cpp`][gamename] and [`stratagus-game-launcher.h`][stratagusgamelauncher] seems to indicate the right location should be `~/.stratagus/data.Stargus`.

So where shoud I put that data folder?

[launcherror]: https://user-images.githubusercontent.com/21345269/33569586-6a4c1b50-d92a-11e7-8de2-aa3df1f2d2c9.png "Stargus data was not extracted yet. You need extract data from original Starcraft CD or installation stardat.mpq or starcraft.mpq file first!"
[installinstructions]: https://github.com/Wargus/stargus/blob/master/README.md#installation-instructions
[gamename]: https://github.com/Wargus/stargus/blob/17e1881f4b7ab5893ba23b9afa0ee4f60bd9231c/stargus.cpp#L20
[stratagusgamelauncher]: https://github.com/Wargus/stratagus/blob/add0ee52651a527d6dbca7edb4544ce1d5b64b35/gameheaders/stratagus-game-launcher.h

**Comment by timfel** [comment]
There are more files that need to be copied to the right directories, just running Startool isn't enough. The status binary should take care of that, the dialog you posted is correct, and upon pressing ok it should prompt you for a stardat.mpq, does that file dialog not open when you press "OK"? (Don't close the dialog with a keyboard shortcut or using the window manager's X button, then it will just cancel)

**Comment by timfel** [comment]
If you want to know how to do the extraction manually, pretty much all you need is in stargus.cpp, including the extra files and folders you need to copy and their target directories.

**Comment by ThibaultLemaire** [comment]
> upon pressing ok it should prompt you for a stardat.mpq, does that file dialog not open when you press "OK"?

No it does not. Stargus simply terminates with exit status 1. I'd like to debug but I'm not very comfortable with C++ and I don't know the appropriate tools.

> If you want to know how to do the extraction manually, pretty much all you need is in stargus.cpp, including the extra files and folders you need to copy and their target directories.

What do you mean? Here's what I've added without success :

- `mpqlist.txt` to `~/.stratagus/data.Stargus/mpqlist.txt`
- `contrib/fog.png` to `~/.stratagus/data.Stargus/graphics/tilesets/fog.png`
- `contrib/fog.png` to `~/.stratagus/data.Stargus/tilesets/fog.png`
- `scripts/` to `~/.stratagus/data.Stargus/scripts/`

**Comment by ThibaultLemaire** [comment]
I'm sorry it looked like I was asking you to get me started on C++. Although I *am* curious about the tools you use, I wouldn't want you to waste any time on teaching me the basics. I mean there are enough resources out there for me to manage on my own.

All I meant was that it might take me some time before I can properly debug and get you more info on this issue.

**Comment by timfel** [comment]
Hi, sorry, I had no time to look into this this week. One thing, could you try launch the game on the cmdline with `stratagus -d ~/.stratagus/data.Stargus` and post the output?

**Comment by ThibaultLemaire** [comment]
`stratagus -d ~/.stratagus/data.Stargus` worked and the game launched.

It looks like I made a mistake while compiling stargus. I compiled against [stratagus][stratagus] not [stratagus-git][stratagusgit] from the AUR. I'm not sure exactly why it behaved like that though.

**So for the record, here's what I've done & tested functional on Arch:**
1. Install [stratagus-git][stratagusgit]
2. Install [stargus-git][stargusgit]
3. Manually generate data into `~/.stratagus/data.Stargus`

By the way, I'd like to write a wiki page giving instructions for manually generating the data as it was not so trivial for me, and other new comers might have the same difficulties making stargus work.

Additionally, I've tried to start afresh with this working installation from the AUR and make it auto-generate the data. This time I got the expected dialog-box which I pointed to `StarDat.mpq` from my wine installation of Starcraft Anthology, but it failed. I'm opening an other issue on that.

With your permission, I'm closing this issue and opening two new ones for the wiki page and the auto-extraction of starcraft data.

[stratagus]: https://aur.archlinux.org/packages/stratagus/
[stratagusgit]: https://aur.archlinux.org/packages/stratagus-git/
[stargusgit]: https://aur.archlinux.org/packages/stargus-git/
